components/molecule/modal/src/index.js

```diff
@@ -0,0 +1,183 @@
+import PropTypes from 'prop-types'
+import React, {Component} from 'react'
+import cx from 'classnames'
+import {SUPPORTED_KEYS} from './config'
+import {suitClass} from './helpers'
+import {Close} from './Close'
+import {HeaderRender} from './HeaderRender'
+import WithAnimation from './HoC/WithAnimation'
+
+const toggleWindowScroll = disableScroll => {
+  window.document.body.classList.toggle('is-MoleculeModal-open', disableScroll)
+}
+
+class MoleculeModal extends Component {
+  _contentRef = React.createRef()
+  _wrapperRef = React.createRef()
+
+  componentDidMount() {
+    document.addEventListener('keydown', this._onKeyDown)
+    this._wrapperRef.current.addEventListener(
+      'webkitAnimationEnd',
+      this.props.onAnimationEnd
+    )
+  }
+  componentWillUnmount() {
+    toggleWindowScroll(false)
+    document.removeEventListener('keydown', this._onKeyDown)
+    this._wrapperRef.current.addEventListener(
```

Comment:
You could just use `https://reactjs.org/docs/events.html#animation-events`.

`onAnimationEnd` on the wrapper div.

Reply:
done!